### PR TITLE
fix: removing com.kjetland dependency and some version conflicts

### PR DESCRIPTION
**Pull request by elbiocaetano**

### CODE REVIEW

**Review by christopher-cudennec — DISMISSED** [review]

File: sda-commons-server-jackson/build.gradle [hunk 1]
```diff
@@ -7,11 +7,5 @@ dependencies {
...
   testImplementation project(':sda-commons-server-testing')
   testImplementation project(':sda-commons-shared-forms')
-  testImplementation 'com.kjetland:mbknor-jackson-jsonschema_2.13', {
-    /**
-     * Dropwizard comes with jakarta.validation-api instead of javax.validation-api.
-     * Both contain the same classes.
-     */
-    exclude group: 'javax.validation', module: 'validation-api'
-  }
+  testImplementation "io.github.classgraph:classgraph"
```
Comment (JoergSiebahn) [inline sda-commons-server-jackson/build.gradle:10]: NIT: I looked up, why we need Classgraph here and found a very picky detail.

The [variable here](https://github.com/SDA-SE/sda-dropwizard-commons/blob/39c053d/sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/errors/JerseyValidationExceptionMapperTest.java#L46) is prefixed with `javax` but references a `jakarta` package.

File: sda-commons-server-openapi/src/test/java/org/sdase/commons/server/openapi/Iso8601SerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.dropwizard.core.Configuration;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import jakarta.ws.rs.client.Invocation;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.sdase.commons.server.openapi.apps.test.OpenApiBundleTestApp;
+
+class Iso8601SerializerTest {
```
Comment (JoergSiebahn) [inline sda-commons-server-openapi/src/test/java/org/sdase/commons/server/openapi/Iso8601SerializerTest.java:15]: NIT: For clarity may be renamed to `Iso8601SerializerOpenAPIDocumentationTest`

File: sda-commons-shared-asyncapi/src/test/java/org/sdase/commons/shared/asyncapi/Iso8601SerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package org.sdase.commons.shared.asyncapi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+class Iso8601SerializerTest {
```
Comment (JoergSiebahn) [inline sda-commons-shared-asyncapi/src/test/java/org/sdase/commons/shared/asyncapi/Iso8601SerializerTest.java:7]: For asyncapi, this is already covered [here](https://github.com/SDA-SE/sda-dropwizard-commons/blob/39c053d/sda-commons-shared-asyncapi/src/test/java/org/sdase/commons/shared/asyncapi/jsonschema/AbstractJsonSchemaBuilderTest.java#L119). Imho we can save some test code here. I assume, `HouseCreated` and the `asyncapi_template.yaml` are also just added for this test?

If you want a dedicated test, I would prefer to test only the schema generator and not the whole async api.

File: sda-commons-dependencies/build.gradle [hunk 4]
```diff
@@ -231,7 +230,7 @@ dependencies {
       because "conflict between io.findify:s3mock_2.13 and org.apache.kafka:kafka_2.13 upgrade"
     }
     api "org.scala-lang:scala-library:$scalaVersion", {
-      because "conflict between org.apache.kafka:kafka_2.13, io.findify:s3mock_2.13 and com.kjetland:mbknor-jackson-jsonschema_2.13:+ upgrade"
+      because "conflict between org.apache.kafka:kafka_2.13 upgrade"
```
[the remark this answers is not in the document]
Comment (christopher-cudennec) [reply sda-commons-dependencies/build.gradle:233]: I'll update the comment. I've also looked into what conflicts are left. At the moment Scala is only used by Kafka and Jackson.

File: sda-commons-server-jackson/build.gradle (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-server-jackson/build.gradle:10]: Eagle eye 🦅 😆 
Fixed it!

File: sda-commons-shared-asyncapi/src/test/java/org/sdase/commons/shared/asyncapi/Iso8601SerializerTest.java (hunk 3 above)
Comment (christopher-cudennec) [reply sda-commons-shared-asyncapi/src/test/java/org/sdase/commons/shared/asyncapi/Iso8601SerializerTest.java:7]: Oh, good to know. I will remove the class and files again! 👍

File: sda-commons-server-cloudevents/build.gradle [hunk 5]
```diff
@@ -1,12 +1,16 @@
 dependencies {
-  api "com.fasterxml.jackson.core:jackson-databind"
-  api "com.kjetland:mbknor-jackson-jsonschema_2.13", {
-    exclude group: "javax.validation", module: "validation-api"
+  api project(':sda-commons-shared-asyncapi'), {
+    exclude group: 'com.fasterxml.jackson.datatype', module: 'jackson-datatype-guava'
+    exclude group: 'com.fasterxml.jackson.module', module: 'jackson-module-blackbird'
+    exclude group: 'com.fasterxml.jackson.jakarta.rs', module: 'jackson-jakarta-rs-json-provider'
```
[the remark this answers is not in the document]
Comment (christopher-cudennec) [reply sda-commons-server-cloudevents/build.gradle:5]: We ran into this only when integration sda-dw-commons into the service repo. I will try to reproduce it there.
I agree, that it looks strange. But version management really is a messy topic that I still don't fully understand. I would've expected that a lot of the conflicts that we saw, should be solved by using `enforcePlatform`

**Pull request merged**